Makefile

```diff
@@ -0,0 +1,58 @@
+# This is used by the image building script referenced below. Normally it just takes the directory name but in this
+# case we want it to be called something else.
+IMAGE_NAME=flytekit-python-template
+VERSION=$(shell ./version.sh)
+
+define PIP_COMPILE
+pip-compile $(1) --upgrade --verbose
+endef
+
+# If the REGISTRY environment variable has been set, that means the image name will not just be tagged as
+#   flytecookbook:<sha> but rather,
+#   docker.io/lyft/flytecookbook:<sha> or whatever your REGISTRY is.
+ifneq ($(origin REGISTRY), undefined)
+	FULL_IMAGE_NAME = ${REGISTRY}/${IMAGE_NAME}
+else
+	FULL_IMAGE_NAME = ${IMAGE_NAME}
+endif
+
+# The Flyte project that we want to register under
+PROJECT=flyteexamples
+# If you want to create a new project, in an environment with flytekit installed run the following:
+# flyte-cli register-project -h localhost:30081 -i - myflyteproject --name "My Flyte Project" \
+#      --description "My very first project getting started on Flyte"
+
+.SILENT: help
+.PHONY: help
+help:
+	echo Available recipes:
+	cat $(MAKEFILE_LIST) | grep -E '^[a-zA-Z0-9_-]+:.*?## .*$$' | awk 'BEGIN { FS = ":.*?## " } { cnt++; a[cnt] = $$1; b[cnt] = $$2; if (length($$1) > max) max = length($$1) } END { for (i = 1; i <= cnt; i++) printf "  $(shell tput setaf 6)%-*s$(shell tput setaf 0) %s\n", max, a[i], b[i] }'
+	tput sgr0
+
+.PHONY: debug
+debug:
+	echo "IMAGE NAME ${IMAGE_NAME}"
+	echo "FULL IMAGE NAME ${FULL_IMAGE_NAME}"
+	echo "VERSION TAG ${VERSION}"
+	echo "REGISTRY ${REGISTRY}"
+
+.PHONY: docker_build
+docker_build:
+	NOPUSH=1 IMAGE_NAME=${IMAGE_NAME} flytekit_build_image.sh ./Dockerfile ${PREFIX}
+
+# The sandbox targets below allow you to upload your code to your hosted Flyte sandbox.
+# Simply run `make register_sandbox` to trigger the sequence.
+.PHONY: in_container_serialize_sandbox
+in_container_serialize_sandbox:
+	pyflyte --config /root/flyte.config serialize workflows -f /tmp/output
```

Comment:
:(, can we avoid the config

Reply:
we have required strings in the config. i don't know how to use pyflyte right now without either passing them as env vars or just passing along the config :\

Reply:
you mean we need some fields and they are in the config only?
That is fine I guess

Reply:
updated flytekit